clients/unshaded/src/main/java/tachyon/client/file/TachyonFileSystem.java

```diff
@@ -0,0 +1,285 @@
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.client.file;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.List;
+
+import tachyon.TachyonURI;
+import tachyon.client.FileSystemMasterClient;
+import tachyon.client.ClientOptions;
+import tachyon.thrift.FileInfo;
+
+/**
+ * Tachyon File System client. This class is the entry point for all file level operations on
+ * Tachyon files. An instance of this class can be obtained via {@link TachyonFileSystem#get}. This
+ * class is thread safe. The read/write interface provided by this client is similar to Java's
+ * input/output streams.
+ */
+public class TachyonFileSystem implements Closeable, TachyonFSCore {
+  /** A cached instance of the TachyonFileSystem */
+  private static TachyonFileSystem sCachedClient;
+
+  /**
+   * @return a TachyonFileSystem instance, there is only one instance available at any time
+   */
+  public static synchronized TachyonFileSystem get() {
+    if (null == sCachedClient) {
+      sCachedClient = new TachyonFileSystem();
+    }
+    return sCachedClient;
+  }
+
+  /** The file system context which contains shared resources, such as the fs master client */
+  private FSContext mContext;
+
+  /**
+   * Constructor, currently TachyonFileSystem does not retain any state
+   */
+  private TachyonFileSystem() {
+    mContext = FSContext.INSTANCE;
+  }
+
+  /**
+   * Closes this TachyonFS instance. The next call to get will create a new TachyonFS instance.
+   * Other references to the old client may still be used.
+   */
+  // TODO: Evaluate the necessity of this method
+  public synchronized void close() {
+    sCachedClient = null;
+  }
+
+  /**
+   * Deletes a file. If the file is a folder, its contents will be deleted recursively. The
+   * delete will abort on a failure, but previous deletes that occurred will still be effective.
+   * The delete will only synchronously be propagated to the master. The file metadata will not
+   * be available after this call, but the data in Tachyon or under storage space may still
+   * reside until the delete is propagated.
+   *
+   * @param file the handler of the file to delete.
+   * @throws IOException if the master is unable to delete the file
+   */
+  @Override
+  public void delete(TachyonFile file) throws IOException {
+    FileSystemMasterClient masterClient = mContext.acquireMasterClient();
+    try {
+      masterClient.deleteFile(file.getFileId(), true);
+    } finally {
```

Comment:
Since this file will the boundary between Tachyon code and user code, better to catch exceptions and LOG.error those exceptions so they show up in Tachyon logs.